include/seqan3/range/container/constexpr_vector.hpp

```diff
@@ -0,0 +1,867 @@
+    }
+
+    /*!\brief Inserts count copies of value before position in the container.
+     * \param   pos Iterator before which the content will be inserted. `pos` may be the end() iterator.
+     * \param count Number of copies.
+     * \param value Element value to insert.
+     * \returns     Iterator pointing to the first element inserted, or `pos` if `count==0`.
+     *
+     * If `size()` + `count` > `capacity()` this function results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Worst-case linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container may contain invalid data after exception is
+     * thrown.
+     */
+    constexpr iterator insert(const_iterator pos, size_type const count, value_type const value)
+    {
+        auto tmp = ranges::view::repeat_n(value, count);
+        return insert(pos, std::ranges::begin(tmp), std::ranges::end(tmp));
+    }
+
+    /*!\brief Inserts elements from range `[begin_it, end_it)` before position in the container.
+     * \tparam begin_iterator_type Must satisfy std::ForwardIterator and the `value_type` must be constructible from
+     *                             the reference type of begin_iterator_type.
+     * \tparam   end_iterator_type Must satisfy std::Sentinel.
+     * \param[in]              pos Iterator before which the content will be inserted. `pos` may be the end() iterator.
+     * \param[in]         begin_it Begin of range to construct/assign from.
+     * \param[in]           end_it End of range to construct/assign from.
+     * \returns                    Iterator pointing to the first element inserted, or `pos` if `begin_it==end_it`.
+     *
+     * The behaviour is undefined if begin_it and end_it are iterators into `*this` or if, given the size `n` of the
+     * range represented by [begin_t, end_it), `size()` + `n` > `capacity()`.
+     *
+     * ### Complexity
+     *
+     * Worst-case linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container may contain invalid data after exception is
+     * thrown.
+     */
+    template <std::ForwardIterator begin_iterator_type, std::Sentinel<begin_iterator_type> end_iterator_type>
+    //!\cond
+        requires std::Constructible<value_type, /*ranges::iter_reference_t*/reference_t<begin_iterator_type>>
+    //!\endcond
+    constexpr iterator insert(const_iterator pos, begin_iterator_type begin_it, end_iterator_type end_it)
+    {
+        auto const pos_as_num = std::ranges::distance(cbegin(), pos);
+        auto const length = std::ranges::distance(begin_it, end_it);
+
+        if (length == 0)
+            return begin(); // nothing to insert
+
+        for (size_type i = sz + length - 1; i > pos_as_num + length - 1; --i)
+            data[i] = data[i - length];
+        // ranges::copy_backward(&data[pos_as_num], &data[sz], &data[sz + length]);
```

Comment:
why not?

Reply:
because although documented as such, it is not constexpr :roll_eyes: 
I'll remove the line.